include/moqt/core/transport.h

```diff
@@ -0,0 +1,226 @@
+            std::map<uint64_t, std::map<uint64_t, std::shared_ptr<BaseTrackHandler>>> pub_tracks_by_name;
+        };
+
+        // -------------------------------------------------------------------------------------------------
+        // Private methods
+        // -------------------------------------------------------------------------------------------------
+        void Init();
+
+        void SendCtrlMsg(const ConnectionContext& conn_ctx, std::vector<uint8_t>&& data);
+        void SendClientSetup();
+        void SendServerSetup(ConnectionContext& conn_ctx);
+        void SendAnnounce(ConnectionContext& conn_ctx, Span<uint8_t const> track_namespace);
+        void SendAnnounceOk(ConnectionContext& conn_ctx, Span<uint8_t const> track_namespace);
+        void SendUnannounce(ConnectionContext& conn_ctx, Span<uint8_t const> track_namespace);
+        void SendSubscribe(ConnectionContext& conn_ctx, uint64_t subscribe_id, FullTrackName& tfn, TrackHash th);
+        void SendSubscribeOk(ConnectionContext& conn_ctx, uint64_t subscribe_id, uint64_t expires, bool content_exists);
+        void SendUnsubscribe(ConnectionContext& conn_ctx, uint64_t subscribe_id);
+        void SendSubscribeDone(ConnectionContext& conn_ctx, uint64_t subscribe_id, const std::string& reason);
+        void SendSubscribeError(ConnectionContext& conn_ctx,
+                                uint64_t subscribe_id,
+                                uint64_t track_alias,
+                                messages::MoqtSubscribeError error,
+                                const std::string& reason);
+        void CloseConnection(TransportConnId conn_id,
+                             messages::MoqtTerminationReason reason,
+                             const std::string& reason_str);
+        bool ProcessRecvCtrlMessage(ConnectionContext& conn_ctx, std::shared_ptr<StreamBuffer<uint8_t>>& stream_buffer);
+        bool ProcessRecvStreamDataMessage(ConnectionContext& conn_ctx,
+                                          std::shared_ptr<StreamBuffer<uint8_t>>& stream_buffer);
+
+        void RemoveSubscribeTrack(ConnectionContext& conn_ctx,
+                                  SubscribeTrackHandler& delegate,
+                                  bool remove_delegate = true);
+
+        std::optional<std::weak_ptr<PublishTrackHandler>> GetPubTrackHandler(ConnectionContext& conn_ctx,
+                                                                             TrackHash& th);
+
+        // -------------------------------------------------------------------------------------------------
+        // Private member variables
+        // -------------------------------------------------------------------------------------------------
+
+        std::mutex state_mutex_;
+        const bool client_mode_;
+        bool stop_{ false };
+        const ServerConfig server_config_;
+        const ClientConfig client_config_;
```

Comment:
these needs to be optional as it is set only in one of the modes. May be I don't understand the need for it to be constants

Reply:
Again, this is not part of the public API as these are private members used by the implementation.